pinot-spi/src/main/java/org/apache/pinot/spi/queryeventlistener/PinotBrokerQueryEventListenerUtils.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.queryeventlistener;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.pinot.spi.annotations.queryeventlistener.BrokerEventListenerFactory;
+import org.apache.pinot.spi.annotations.queryeventlistener.BrokerQueryEventListenerFactory;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.PinotReflectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.pinot.spi.utils.CommonConstants.CONFIG_OF_EVENT_LISTENER_FACTORY_CLASS_NAME;
+import static org.apache.pinot.spi.utils.CommonConstants.DEFAULT_EVENT_LISTENER_FACTORY_CLASS_NAME;
+
+public class PinotBrokerQueryEventListenerUtils {
+    private PinotBrokerQueryEventListenerUtils() {
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PinotBrokerQueryEventListenerUtils.class);
+    private static final String EVENT_LISTENER_PACKAGE_REGEX_PATTERN = ".*\\.plugin\\.query.event.listener\\..*";
```

Comment:
Why do we need reflection here? The class should be loaded by the class name within the config

Reply:
Yeah good point! Will remove.